src/controllers/router-store/test.tsx

```diff
@@ -1,4 +1,5 @@
 import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```

Comment:
Why not importing this in the setup file? So we don't get strange errors if we move tests around

Reply:
I tried moving in my previous PR but the TS compiler starts complaining and no matter what I tried it would not satisfy the error.

Reply:
Bigger significant changes are needed to the config, and TSConfig is required. Basing this on all the simple changes I tried. 

Might be better in another PR as this one already has a lot of changes